src/perl/lib/wtsi_clarity/epp/isc/analyte_pooler.pm

```diff
@@ -94,16 +94,18 @@ sub _uniq_array {
   my ($self, @array) = @_;
 
   my %seen;
-  return grep { !$seen{$_}++ } @array;
+  my @uniq_array =  grep { !$seen{$_}++ } @array;
+
+  return \@uniq_array;
 }
 
 sub _get_values_from_nodelist {
```

Comment:
I thought, I was deleted it from here, when I extracted it to a role. I think it should stay private, because it will be a part of a mixin...